app/hooks/useSearch/index.ts

```diff
@@ -1,25 +1,42 @@
-import { useState, useEffect } from 'react';
+import { useState, useEffect } from 'react'
 
-import { search, SearchResultType } from '~/models/entity.client';
+import { search, SearchResultType } from '~/models/entity.client'
 
 export const useSearch = () => {
-  const [query, setQuery] = useState('');
-  const [results, setResult] = useState<SearchResultType>({
-    artist: [],
-    audio: [],
-    collection: [],
-  });
+  const [query, setQuery] = useState('')
+  const [results, setResult] = useState<
+    | { type: 'searchIt'; res: SearchResultType }
+    | {
+        res: SearchResultType
+        type: 'resolve'
+      }
+    | { res: SearchResultType; type: 'pending' }
+    | { res: SearchResultType; type: 'notFound' }
```

Comment:
Aside from the recommendation for not adding these states in this PR, I recommend you to
Never add types or interfaces inline. They belong to `types.ts`. And if, in an extremely rare case, we wanted to define them in this file, we should define them outside the hook with a name, and refer to them here.
For this PR, drop the states.